src/chi-vue/src/components/data-table/DataTable.tsx

```diff
@@ -748,38 +749,39 @@ export default class DataTable extends Vue {
 
   mounted() {
     const dataTableComponent = this.$refs.dataTable as HTMLElement;
-
     if (dataTableComponent && this.config.columnResize) {
       this._generateColumnResize(dataTableComponent);
     }
 
-    (this.$refs.pagination as Vue).$on(PAGINATION_EVENTS.PAGE_SIZE, (ev: string) => {
-      const data = this._sortedData && this._sortedData.length > 0 ? this._sortedData : this._serializedDataBody;
-
-      if (ev === 'all') {
-        this.resultsPerPage = this._serializedDataBody.length;
-      } else {
-        this.resultsPerPage = parseInt(ev);
-      }
-      this.slicedData = this.sliceData(data);
-      this.$emit(PAGINATION_EVENTS.PAGE_SIZE, this.slicedData);
-    });
-
-    (this.$refs.pagination as Vue).$on(PAGINATION_EVENTS.PAGE_CHANGE, (ev: number) => {
-      const data = this._sortedData && this._sortedData.length > 0 ? this._sortedData : this._serializedDataBody;
-      const numberOfPages = this._calculateNumberOfPages();
-      const pageChangeEventData: DataTablePageChange = {
-        page: ev,
-        data: this.slicedData,
-      };
+    if (this.$refs.pagination) {
+      (this.$refs.pagination as Vue).$on(PAGINATION_EVENTS.PAGE_SIZE, (ev: string) => {
+        const data = this._sortedData && this._sortedData.length > 0 ? this._sortedData : this._serializedDataBody;
 
-      if (ev >= 1 && ev <= numberOfPages) {
-        this.currentPage = ev;
+        if (ev === 'all') {
+          this.resultsPerPage = this._serializedDataBody.length;
+        } else {
+          this.resultsPerPage = parseInt(ev);
+        }
         this.slicedData = this.sliceData(data);
-        this.$emit(PAGINATION_EVENTS.PAGE_CHANGE, pageChangeEventData);
-        this._checkSelectAllCheckbox();
-      }
-    });
+        this.$emit(PAGINATION_EVENTS.PAGE_SIZE, this.slicedData);
+      });
+
+      (this.$refs.pagination as Vue).$on(PAGINATION_EVENTS.PAGE_CHANGE, (ev: number) => {
+        const data = this._sortedData && this._sortedData.length > 0 ? this._sortedData : this._serializedDataBody;
+        const numberOfPages = this._calculateNumberOfPages();
+        const pageChangeEventData: DataTablePageChange = {
+          page: ev,
+          data: this.slicedData,
```

Comment:
If mode is serverSidePaginated, data should be empty. I guess it will be empty indeed, right?

Reply:
@janibolkvadze explained me and it works as expected. Thanks Jani.